pkg/storage/bulk/sst_batcher.go

```diff
@@ -122,6 +147,16 @@ func (b *SSTBatcher) AddMVCCKey(ctx context.Context, key engine.MVCCKey, value [
 	if err := b.rowCounter.Count(key.Key); err != nil {
 		return err
 	}
+
+	// If we do not allowing shadowing of keys when ingesting an SST via
+	// AddSSTable, then we can update the MVCCStats on the fly because we are
+	// guaranteed to ingest unique keys. This saves us an extra iteration in
+	// AddSSTable which has been identified as a significant performance
+	// regression for IMPORT.
+	if b.disallowShadowing {
+		b.updateMVCCStats(&key, &value)
```

Comment:
is escape analysis smart enough to recognize that taking these pointers don't need to move key and value to heap here? I don't know if we need to pass pointers?